packages/aws-cdk-lib/aws-rds/lib/aurora-cluster-instance.ts

```diff
@@ -450,11 +450,13 @@ class AuroraClusterInstance extends Resource implements IAuroraClusterInstance {
 
     const isOwnedResource = Resource.isOwnedResource(props.cluster);
     let internetConnected;
-    let publiclyAccessible = props.publiclyAccessible;
+    let publiclyAccessible;
     if (isOwnedResource) {
       const ownedCluster = props.cluster as DatabaseCluster;
       internetConnected = ownedCluster.vpc.selectSubnets(ownedCluster.vpcSubnets).internetConnectivityEstablished;
-      publiclyAccessible = ownedCluster.vpcSubnets && ownedCluster.vpcSubnets.subnetType === ec2.SubnetType.PUBLIC;
+      publiclyAccessible = props.publiclyAccessible ?? (ownedCluster.vpcSubnets && ownedCluster.vpcSubnets.subnetType === ec2.SubnetType.PUBLIC);
```

Comment:
```suggestion
      const isInPublicSubnet = ownedCluster.vpcSubnets && ownedCluster.vpcSubnets.subnetType === ec2.SubnetType.PUBLIC;
      publiclyAccessible = isInPublicSubnet && (props.publiclyAccessible ?? true);
```
I don't think a public IP is needed if the instance is in a non-public subnet.

Reply:
This will be contrary to the behavior indicated in the documentation here:
https://github.com/aws/aws-cdk/tree/main/packages/aws-cdk-lib/aws-rds#setting-public-accessibility

I believe the correct behavior here is to honor the user's setting if provided as a `publiclyAccessible ` property. If no value is provided we can default based on the interpreted intent from the subnet placement.